src/ToEnum.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter;
+
+use BackedEnum;
+use Laminas\Filter\Exception\RuntimeException;
+use Laminas\Stdlib\ArrayUtils;
+use Traversable;
+use UnitEnum;
+
+use function is_array;
+use function is_int;
+use function is_string;
+use function is_subclass_of;
+
+/**
+ * @psalm-type Options array{enum: class-string<UnitEnum>}
+ * @extends AbstractFilter<Options>
+ */
+class ToEnum extends AbstractFilter
+{
+    /** @var Options */
+    protected $options = [
+        'enum' => null,
+    ];
+
+    /**
+     * @param class-string<UnitEnum>|Traversable|Options $enumOrOptions
+     */
+    public function __construct($enumOrOptions)
+    {
+        if ($enumOrOptions instanceof Traversable) {
+            $enumOrOptions = ArrayUtils::iteratorToArray($enumOrOptions);
+        }
+
+        if (
+            is_array($enumOrOptions) &&
+            isset($enumOrOptions['enum'])
+        ) {
+            $this->setOptions($enumOrOptions);
+
+            return;
+        }
+
+        if (is_string($enumOrOptions)) {
+            $this->setEnum($enumOrOptions);
+        }
+    }
+
+    /**
+     * @param class-string<UnitEnum> $enum
+     */
+    public function setEnum(string $enum): self
```

Comment:
IMO `setEnum()` to be dropped or made `protected`: shouldn't be called from public scope